### PR TITLE
chore(emotion): Convert badge to styled components

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/badge.tsx [hunk 1]
```diff
@@ -1,16 +1,41 @@
...
 import React from 'react';
-import {cx} from 'react-emotion';
+import styled from 'react-emotion';
+import space from 'app/styles/space';
+import theme from 'app/utils/theme';
 
-type Props = {
+type Props = React.HTMLProps<HTMLSpanElement> & {
   text?: string;
   priority?: 'strong' | 'new' | 'highlight';
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/badge.tsx]: Could this be `keyof priorityColors`?

File: src/sentry/static/sentry/app/components/badge.tsx [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
   className?: string;
 };
 
-const Badge = ({priority, className, text}: Props) => (
-  <span className={cx('badge', priority, className)}>{text}</span>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/badge.tsx]: do we have any tests that rely on `badge` as a className?

File: src/sentry/static/sentry/app/components/badge.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/badge.tsx]: yea

File: src/sentry/static/sentry/app/components/badge.tsx (hunk 2 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/badge.tsx]: Not as far as I can tell, they all seem to be passing anyway

**Review by billyvg — APPROVED** [review]

**Pull request merged**